src/confab/agent/runner.go

```diff
@@ -58,12 +61,19 @@ func (r *Runner) Run() error {
 		return err
 	}
 
-	go r.cmd.Wait() // reap child process if it dies
+	r.wg.Add(1)
+	go func() {
+		r.cmd.Wait()
+		atomic.StoreInt32(&r.exited, 1)
+		r.wg.Done()
+	}()
 
 	r.Logger.Info("agent-runner.run.success")
 	return nil
 }
 
+func (r *Runner) Exited() bool { return atomic.LoadInt32(&r.exited) == 1 }
```

Comment:
The implementation of `os.Process.Wait` provides a stronger and better tested assertion that the process is running, and is cross-platform (unix: [wait4](http://linux.die.net/man/2/wait4) windows: [ WaitForSingleObject](https://msdn.microsoft.com/en-us/library/windows/desktop/ms687032(v=vs.85).aspx)).  Since a call to `Wait` must occur after a call to `Run` I don't see the value in polling (we know the process either started or failed to start) when the standard library and OS provides the ability to wait for a process to exit.  

Additionally, we should probably by checking the error returned by `Wait`, this was being discarded completely in the previous implementation.